File: 1122/02/snippet.py
#
# 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
#
# 
# @param tokens string字符串一维数组 
# @return int整型
#
class Solution:
    def evalRPN(self , tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i in "+-*/":
                a = stack.pop()
                b = stack.pop()
                if i == '+':
                    stack.append(b + a)
                elif i == '-':
                    stack.append(b - a)
                elif i == '*':
                    stack.append(b * a)
                elif i == '/':
                    stack.append(int(b / a))
            else:
                stack.append(int(i))
        return stack[0]

File: 1122/02/snippet.cpp
class Solution {
public:
    /**
     * 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
     *
     * 
     * @param tokens string字符串vector 
     * @return int整型
     */
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for(int i = 0; i < tokens.size(); i++){
            if(tokens[i] == "+" || tokens[i] == "-" || tokens[i] == "*" || tokens[i] == "/"){
                int n1 = st.top();
                st.pop();
                int n2 = st.top();
                st.pop();
                if(tokens[i] == "+")
                st.push(n2 + n1);
                if(tokens[i] == "-")
                st.push(n2 - n1);
                if(tokens[i] == "*")
                st.push(n2 * n1);
                if(tokens[i] == "/")
                st.push(n2 / n1);
            }else{
                int k = stoi(tokens[i]);
                st.push(k);
            }
        }
        return st.top();
    }
};